### ISSUE TITLE
Task Module not working in ios teams bot

### ISSUE THREAD

**Issue by Rajkumar235216** [microsoft/BotBuilder-Samples#3778]
## Sample information
1. Sample type: samples
2. Sample language: python
3. Sample name: 54. teams-task-module

## Describe the bug
Task module not working when opened using Action.Submit button in ios teams chatbot(personal scope)

## To Reproduce
Cloned the repo
Navigated to folder 54. teams-task-module
Did npm install
Updated .env file and manifest json with app ids
Created a zip and installed the app ms teams personal scope.
Sent a message after installation and bot replied with hero card and adaptive card.
Tried pressing with one of the buttons in adaptive cards in ios mobile teams app bot and it says:
 "The bot encountered an error or bug."
"To continue to run this bot, please fix the bot source code."
I tried printing traceback of error on bot console. Got following traceback error:

Traceback (most recent call last):
  File "/home/ec2-user/azurebot/lib64/python3.6/dist-packages/msrest/serialization.py", line 1418, in _deserialize
    found_value = key_extractor(attr, attr_desc, data)
  File "/home/ec2-user/azurebot/lib64/python3.6/dist-packages/msrest/serialization.py", line 1162, in rest_key_extractor
    return working_data.get(key)
AttributeError: 'str' object has no attribute 'get'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ec2-user/azurebot/lib64/python3.6/dist-packages/botbuilder/core/bot_adapter.py", line 129, in run_pipeline
    context, callback
  File "/home/ec2-user/azurebot/lib64/python3.6/dist-packages/botbuilder/core/middleware_set.py", line 69, in receive_activity_with_status
    return await self.receive_activity_internal(context, callback)
  File "/home/ec2-user/azurebot/lib64/python3.6/dist-packages/botbuilder/core/middleware_set.py", line 79, in receive_activity_internal
    return await callback(context)
  File "/home/ec2-user/azurebot/lib64/python3.6/dist-packages/botbuilder/core/activity_handler.py", line 78, in on_turn
    invoke_response = await self.on_invoke_activity(turn_context)
  File "/home/ec2-user/azurebot/lib64/python3.6/dist-packages/botbuilder/core/teams/teams_activity_handler.py", line 155, in on_invoke_activity
    TaskModuleRequest, turn_context.activity.value
  File "/home/ec2-user/azurebot/lib64/python3.6/dist-packages/botbuilder/core/serializer_helper.py", line 28, in deserializer_helper
    return deserializer(msrest_cls.__name__, dict_to_deserialize)
  File "/home/ec2-user/azurebot/lib64/python3.6/dist-packages/msrest/serialization.py", line 1358, in __call__
    return self._deserialize(target_obj, data)
  File "/home/ec2-user/azurebot/lib64/python3.6/dist-packages/msrest/serialization.py", line 1432, in _deserialize
    value = self.deserialize_data(raw_value, attr_desc['type'])
  File "/home/ec2-user/azurebot/lib64/python3.6/dist-packages/msrest/serialization.py", line 1601, in deserialize_data
    return self._deserialize(obj_type, data)
  File "/home/ec2-user/azurebot/lib64/python3.6/dist-packages/msrest/serialization.py", line 1436, in _deserialize
    raise_with_traceback(DeserializationError, msg, err)
  File "/home/ec2-user/azurebot/lib64/python3.6/dist-packages/msrest/exceptions.py", line 51, in raise_with_traceback
    raise error.with_traceback(exc_traceback)
  File "/home/ec2-user/azurebot/lib64/python3.6/dist-packages/msrest/serialization.py", line 1418, in _deserialize
    found_value = key_extractor(attr, attr_desc, data)
  File "/home/ec2-user/azurebot/lib64/python3.6/dist-packages/msrest/serialization.py", line 1162, in rest_key_extractor
    return working_data.get(key)
msrest.exceptions.DeserializationError: Unable to deserialize to object: type, AttributeError: 'str' object has no attribute 'get'

## Expected behavior
Task module should work in ios teams chatbot(personal scope) as it is working in windows/Mac desktops/web team app chatbot.

**Comment by ramfattah** [comment]
Hello @Rajkumar235216, I'm investigating this one.

**Comment by Rajkumar235216** [comment]
Hi @ramfattah 
Is there any update on this issue?

**Comment by ramfattah** [comment]
Hi @Rajkumar235216, thanks for following up. 
We are actively investigating this issue. Will report back once we have an update.

Thanks.

**Comment by Rajkumar235216** [comment]
Hi @ramfattah 
Can you try adding domains under  manifest.json -> validDomains without prepending with "https://".
for example, if we want to add url of microsoft then instead of adding as "https://microsoft.com",  we can add "microsoft.com".

Please let me know, if that works here for you.

For me, these steps are working fine in Mac OS, Windows, Web and Android devices. Except for IOS devices where error also different from here as mentioned in "To Reproduce" section in this Reported issue description.

**Comment by ramfattah** [comment]
Hi @Rajkumar235216,

Tried to add the domain in manifest.json without prepending `https://` and still, the same behavior occurs of HTML pages not getting rendered. 

It appears I'm reproducing open issue #3695. Not sure if it's related to this issue.

Investigating further. I will report back once we have an update. 
Thank you.

**Comment by ramfattah** [comment]
Hi @Rajkumar235216 

Update: I'm now unblocked from #3695, the sample is working fine from my machine.
The next step would be to test [54.teams-task-module](https://github.com/microsoft/BotBuilder-Samples/tree/main/samples/python/54.teams-task-module) python sample in IOS to see if we can get a repro.

We will report back once we have an update. Thanks for your patience.

**Comment by ramfattah** [comment]
Hi @Rajkumar235216,

I'm able to reproduce this issue using python sample [54.teams-task-module](https://github.com/microsoft/BotBuilder-Samples/tree/main/samples/python/54.teams-task-module).


#### Here are the steps I took:
1) Ran sample [54.teams-task-module](https://github.com/microsoft/BotBuilder-Samples/tree/main/samples/python/54.teams-task-module) using ngrok
2) Created [Bot Framework registration resource](https://docs.microsoft.com/en-us/azure/bot-service/bot-service-quickstart-registration) in Azure
     * Used the https URL given by running ngrok
3) Installed the bot app in Teams using the following [documentation](https://github.com/microsoft/BotBuilder-Samples/tree/main/samples/python/54.teams-task-module#:~:text=This%20step%20is,a%20custom%20app%22)
3) Interact with the bot in Teams using iPhone
     * Send any activity
     * Tap on `Custom Form` or `Youtube` button
4) see the error terminal 

#

Looks like when using iPhone Teams app, after clicking the `Custom Form` or `Youtube` button, the following error is thrown:
```
 [on_turn_error] unhandled error: Unable to deserialize to object: type, AttributeError: 'str' object has no attribute 'get'
Traceback (most recent call last):
  File "F:\BotBuilder-Samples\samples\python\54.teams-task-module\env\lib\site-packages\msrest\serialization.py", line 1418, in _deserialize
    found_value = key_extractor(attr, attr_desc, data)
  File "F:\BotBuilder-Samples\samples\python\54.teams-task-module\env\lib\site-packages\msrest\serialization.py", line 1162, in rest_key_extractor
    return working_data.get(key)
AttributeError: 'str' object has no attribute 'get'

```

**Comment by Rajkumar235216** [comment]
Hi @ramfattah, Thanks for update.
Yes, I am getting same error in iPhone.
Please do the needful in resolving this error.

**Comment by Wajeed-msft** [comment]
@Rajkumar235216  - I have copied this over to Microsoft Teams repo. Please use [this GitHub issue ](https://github.com/MicrosoftDocs/msteams-docs/issues/6435) for tracking.

**Comment by ramfattah** [comment]
Closing as this transferred to [msteams-docs](https://github.com/MicrosoftDocs/msteams-docs) repo.
New issue tracking here https://github.com/MicrosoftDocs/msteams-docs/issues/6435
